library/core/src/iter/range.rs

```diff
@@ -494,6 +502,13 @@ impl Step for char {
         Some(unsafe { char::from_u32_unchecked(res) })
     }
 
+    #[requires((start as u32).checked_add(count as u32).is_some())]
+    #[requires({
+        let dist = (start as u32).checked_add(count as u32);
+        dist.is_some() &&
+            ((start as u32) >= 0xD800 || dist.unwrap() < 0xD800 ||
+             dist.unwrap().checked_add(0x800).is_some())
```

Comment:
Nit: use `is_some_and`
```suggestion
        (start as u32).checked_add(count as u32).is_some_and(|dist|
            (start as u32) >= 0xD800 || 
            dist < 0xD800 ||
            dist.checked_add(0x800).is_some()
         )
```